openpype/hosts/flame/api/plugin.py

```diff
@@ -1,3 +1,618 @@
+                headline.addWidget(Spacer(20, self))
+                headline.addWidget(QtWidgets.QLabel(v["label"]))
+
+                # adding nested layout with label
+                self.content_layout.append(QtWidgets.QWidget(self))
+                self.content_layout[-1].setObjectName("sectionContent")
+
+                nested_content_layout = QtWidgets.QFormLayout(
+                    self.content_layout[-1])
+                nested_content_layout.setObjectName("NestedContentLayout")
+                content_layout.addWidget(self.content_layout[-1])
+
+                # add nested key as label
+                data[k]["value"] = self.populate_widgets(
+                    v["value"], nested_content_layout)
+
+            elif v["type"] == "QLineEdit":
+                data[k]["value"] = self.create_row(
+                    content_layout, "QLineEdit", v["label"],
+                    setText=v["value"], setToolTip=tool_tip)
+            elif v["type"] == "QComboBox":
+                data[k]["value"] = self.create_row(
+                    content_layout, "QComboBox", v["label"],
+                    addItems=v["value"], setToolTip=tool_tip)
+            elif v["type"] == "QCheckBox":
+                data[k]["value"] = self.create_row(
+                    content_layout, "QCheckBox", v["label"],
+                    setChecked=v["value"], setToolTip=tool_tip)
+            elif v["type"] == "QSpinBox":
+                data[k]["value"] = self.create_row(
+                    content_layout, "QSpinBox", v["label"],
+                    setValue=v["value"], setMinimum=0,
+                    setMaximum=100000, setToolTip=tool_tip)
+        return data
+
+
+class Spacer(QtWidgets.QWidget):
+    def __init__(self, height, *args, **kwargs):
+        super(self.__class__, self).__init__(*args, **kwargs)
+
+        self.setFixedHeight(height)
+
+        real_spacer = QtWidgets.QWidget(self)
+        real_spacer.setObjectName("Spacer")
+        real_spacer.setFixedHeight(height)
+
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.setContentsMargins(0, 0, 0, 0)
+        layout.addWidget(real_spacer)
+
+        self.setLayout(layout)
+
+
+class Creator(openpype.Creator):
+    """Creator class wrapper
+    """
+    clip_color = lib.ctx.color_map["purple"]
+    rename_index = None
+
+    def __init__(self, *args, **kwargs):
+        super(Creator, self).__init__(*args, **kwargs)
+        self.presets = openpype.get_current_project_settings()[
+            "flame"]["create"].get(self.__class__.__name__, {})
+
+        # adding basic current context flame objects
+        self.project = lib.get_current_project()
+        self.sequence = lib.get_current_sequence(opflame.selection)
+
+        if (self.options or {}).get("useSelection"):
+            self.selected = lib.get_sequence_segments(self.sequence, True)
+        else:
+            self.selected = lib.get_sequence_segments(self.sequence)
+
+        self.widget = CreatorWidget
+
```

Comment:
I would recommend to not store class as attribute and then call it in code but rather implement method which does that on call and gives ability to override it if needed (this would require changes in Creators).
```suggestion
def _create_widget(self, *args, **kwargs):
        return CreatorWidget(*args, **kwargs)

```

Reply:
thanks, implemented in gitkraken://repolink/33c731ef76967f4bd3f6ba8b287ecc0c52412f33/commit/389e90670ed5181d9dd45211950f67f9aec1cc5e?url=https%3A%2F%2Fgithub.com%2Fpypeclub%2FOpenPype.git